File: server/API/Models/Suggestion.cs
namespace VoterAPI.Models;

public class Suggestion
{
    public int Id { get; set; }
    public int BoardId { get; set; }
    public required string Text { get; set; }
    public int SubmittedByUserId { get; set; }
    public DateTime SubmittedDate { get; set; } = DateTime.UtcNow;
    public SuggestionStatus Status { get; set; } = SuggestionStatus.Pending;
    public bool IsVisible { get; set; } = true;

    // Navigation properties
    public Board? Board { get; set; }
    public User? SubmittedBy { get; set; }
    public ICollection<Vote> Votes { get; set; } = new List<Vote>();
}

File: client/src/types/suggestion.ts
export interface Suggestion {
  id: number;
  boardId: number;
  text: string;
  submittedByUserId: number;
  submittedByUsername?: string;
  submittedDate: string;
  status: 'Pending' | 'Approved' | 'Rejected';
  isVisible: boolean;
  voteCount: number;
  userHasVoted: boolean;
}

export interface CreateSuggestionRequest {
  boardId: number;
  text: string;
}
